django/core/management/commands/compilemessages.py

```diff
@@ -121,30 +122,33 @@ def compile_messages(self, locations):
         with concurrent.futures.ThreadPoolExecutor() as executor:
             futures = []
             for i, (dirpath, f) in enumerate(locations):
+                po_path = Path(dirpath) / f
+                mo_path = po_path.with_suffix('.mo')
+                if mo_path.exists() and mo_path.stat().st_mtime >= po_path.stat().st_mtime:
```

Comment:
Would it be cheaper to call `mo_path.stat()` unconditionally and catch `FileNotFoundError`?

Reply:
Not sure either, I was throwing it out there it's likely faster to build an exception on the Python side that perform I/O if unnecessary.

Reply:
There is in theory a chance of hitting a TOCTTOU case here anyway, in which case it is better to handle the exception to avoid a crash.